presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -62,50 +109,84 @@ public PlanNodeStatsEstimate calculateStats(PlanNode node, StatsProvider sourceS
         return getStatistics(node, session, lookup, delegateStats);
     }
 
+    @Override
+    public void registerPlan(PlanNode root, Session session)
+    {
+        // Only precompute history based stats when plan has a join/aggregation.
+        if (!PlanNodeSearcher.searchFrom(root).where(node -> PRECOMPUTE_PLAN_NODES.stream().anyMatch(clazz -> clazz.isInstance(node))).matches()) {
+            return;
+        }
+        ImmutableList.Builder<PlanNodeWithHash> planNodesWithHash = ImmutableList.builder();
+        forTree(PlanNode::getSources).depthFirstPreOrder(root).forEach(plan -> {
+            if (plan.getStatsEquivalentPlanNode().isPresent()) {
+                planNodesWithHash.addAll(getPlanNodeHashes(plan, session).values());
+            }
+        });
+        try {
+            cache.getAll(planNodesWithHash.build());
+        }
+        catch (ExecutionException e) {
+            LOG.error("Unable to register plan %s", e);
```

Comment:
Here we swallow the exception. Would it cause any issue of the caller? I see line 64 in. HistoricalStatisticsEquivalentPlanMarkingOptimizer.java doesn't check anything by calling this method.

Reply:
If we can't find history, we return empty statistics in other parts of this class. Here, we just care about caching so we ignore it

Reply:
Maybe a warning is better than throwing. Missing history stats should not fail the query.